### DISCUSSION TITLE
Consider using a code coverage service

### DISCUSSION THREAD

**Discussion by bouweandela** [SciTools/iris#5014] [Ideas]
At the moment, it is not clear to me as a contributor which parts of the code that I am contributing are covered by tests, so I don't know if can safely change things or if I'm silently breaking something. I guess the same goes for pull request reviewers.

It would be convenient to
1) Explain in the contribution guidelines how to generate code coverage metrics
2) Use a code coverage service that reports code coverage (changes) on every pull request. Popular examples of such services are [Codecov](https://about.codecov.io/) and [Coveralls](https://coveralls.io/)

@SciTools/iris-devs Is this something you have considered before? Does it sound interesting?

**Comment by trexfeathers** [comment]
#4765 

@lbdreyer

**Reply by bouweandela** [reply]
That looks great! Are there any plans to finalize that pull request?
